development-docs/systemtests/io.strimzi.systemtest.cruisecontrol.CruiseControlST.md

```diff
@@ -0,0 +1,218 @@
+| - | - | - |
+| 1. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 2. | Create Kafka cluster with Cruise Control | Kafka cluster with Cruise Control is created and running |
+| 3. | Create KafkaRebalance resource | KafkaRebalance resource is created and running |
+| 4. | Wait until KafkaRebalance is in ProposalReady state | KafkaRebalance reaches ProposalReady state |
+| 5. | Annotate KafkaRebalance with 'approve' | KafkaRebalance is annotated with approval |
+| 6. | Update KafkaRebalance spec to configure replication throttle | KafkaRebalance resource's spec is updated |
+| 7. | Wait until KafkaRebalance returns to ProposalReady state | KafkaRebalance re-enters ProposalReady state following the update |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlDuringBrokerScaleUpAndDown
+
+**Description:** Testing the behavior of Cruise Control during both scaling up and down of Kafka brokers using node pools.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 2. | Create initial Kafka cluster setup with Cruise Control and topic | Kafka cluster, topic, and scraper pod are created successfully. |
+| 3. | Scale Kafka up to a higher number of brokers | Kafka brokers are scaled up to the specified number of replicas. |
+| 4. | Create a KafkaRebalance resource with add_brokers mode | KafkaRebalance proposal is ready and processed for adding brokers. |
+| 5. | Check the topic's replicas on the new brokers | Topic has replicas on one of the newly added brokers. |
+| 6. | Create a KafkaRebalance resource with remove_brokers mode | KafkaRebalance proposal is ready and processed for removing brokers. |
+| 7. | Check the topic's replicas only on initial brokers | Topic replicas are only on the initial set of brokers. |
+| 8. | Scale Kafka down to the initial number of brokers | Kafka brokers are scaled down to the original number of replicas. |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlIntraBrokerBalancing
+
+**Description:** Test ensuring the intra-broker disk balancing with Cruise Control works as expected.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Initialize JBOD storage configuration | JBOD storage with specific disk sizes are initialized |
+| 2. | Create Kafka broker and controller pools using the initialized storage | Kafka broker and controller pools are created and available |
+| 3. | Deploy Kafka with Cruise Control enabled | Kafka deployment with Cruise Control is successfully created |
+| 4. | Create Kafka Rebalance resource with disk rebalancing configured | Kafka Rebalance resource is created and configured for disk balancing |
+| 5. | Wait for the Kafka Rebalance to reach the 'ProposalReady' state | Kafka Rebalance resource reaches the 'ProposalReady' state |
```

Comment:
In most places ProposalReady is not wrapped in quotes, Maybe remove from here (or add elsewhere). perhaps check other states too.

Reply:
Yeah, true. I will remove it to make it consistent